LayoutTests/http/wpt/2dcontext/imagebitmap/noisy-apis.html

```diff
@@ -0,0 +1,51 @@
+<!DOCTYPE html>
+<title>Canvas test: noisy APIs</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/html/canvas/resources/canvas-tests.js"></script>
+<script src="common.js"></script>
+<body>
+<script>
+String.prototype.hash = function() {
+  var hash = 0, i, chr;
+  if (this.length === 0) return hash;
+  for (i = 0; i < this.length; i++) {
+    chr   = this.charCodeAt(i);
+    hash  = ((hash << 5) - hash) + chr;
+    hash |= 0; // Convert to 32bit integer
+  }
+  return hash;
+};
+
+(function() {
+    promise_test(function() {
+        return new Promise(function(resolve, reject) {
+            let canvas = createCanvasOfSize(10, 10);
+            initializeSolidCanvas(canvas, 255,0,0,255);
+            resolve(canvas);
+        }).then(function(canvas) {
+            _assert(testCanvasPixelsDifferent(canvas, 255,0,0,255), "testCanvasPixelsDifferent");
```

Comment:
This will fail about 2^-32 of the time.  That may be something we don't care about, but adding a function to Internals.idl that calls setDomainIdentifier with a known deterministic value would prevent that and also allow us to measure the exact values of the pixels instead of just testing that they are close to but not equal to this value.